terraform/aws/security/aws-dynamodb-table-unencrypted.tf

```diff
@@ -0,0 +1,72 @@
+# pass
+
+resource "aws_dynamodb_table" "cmk" {
+  name           = "GameScores"
+  billing_mode   = "PROVISIONED"
+  read_capacity  = 20
+  write_capacity = 20
+  hash_key       = "UserId"
+  range_key      = "UserId"
+
+  attribute {
+    name = "UserId"
+    type = "S"
+  }
+
+  server_side_encryption {
+      enabled = true
+      kms_key_arn = "arn:aws:kms:us-west-2:123456789012:key/1234abcd-12ab-34cd-56ef-1234567890ab"
+  }
+}
+
+# failure
+# ruleid: aws-dynamodb-table-unencrypted
+resource "aws_dynamodb_table" "default" {
+  name           = "GameScores"
+  billing_mode   = "PROVISIONED"
+  read_capacity  = 20
+  write_capacity = 20
+  hash_key       = "UserId"
+  range_key      = "UserId"
+
+  attribute {
+    name = "UserId"
+    type = "S"
+  }
+}
+# ruleid: aws-dynamodb-table-unencrypted
+resource "aws_dynamodb_table" "encrypted_false" {
+  name           = "GameScores"
+  billing_mode   = "PROVISIONED"
+  read_capacity  = 20
+  write_capacity = 20
+  hash_key       = "UserId"
+  range_key      = "UserId"
+
+  attribute {
+    name = "UserId"
+    type = "S"
+  }
+
+  server_side_encryption {
+      enabled = false
+  }
+}
+
+resource "aws_dynamodb_table" "encrypted_no_cmk" {
+  name           = "GameScores"
+  billing_mode   = "PROVISIONED"
+  read_capacity  = 20
+  write_capacity = 20
+  hash_key       = "UserId"
+  range_key      = "UserId"
+
+  attribute {
+    name = "UserId"
+    type = "S"
+  }
+
+  server_side_encryption {
+      enabled = true
```

Comment:
This one will still get flagged because it does not contain the `kms_key_arn = ...` inside. Do you want to flag this one, or does the pattern need amending?

Reply:
Yes, since it has no cmk, we want to flag this one